### DISCUSSION TITLE
hostname_replace for youtube

### DISCUSSION THREAD

**Discussion by caywin25** [searxng/searxng#1922] [Q&A]
I just set hostname_replace for youtube links to: piped.kavin.rocks

```  
'(.*\.)?youtube\.com$': 'piped.kavin.rocks'
'(.*\.)?youtu\.be$': 'piped.kavin.rocks'
'(.*\.)?youtube-noocookie\.com$': 'piped.kavin.rocks'
```

It works fine if I click on the link but when I want to play the video in thumbnail, youtube is launched.

Is this normal or a bug?

Thanks

**Comment by return42** [comment]
Can't say why it does not works for you, but for me the link on the thumb works: https://darmarit.org/searx/search?q=%21youtube+maceo+parker 

BTW: here are my hostname_replace setting:

- https://github.com/searxng/searxng/discussions/911
- https://github.com/searxng/searxng/discussions/970

**Reply by caywin25** [reply]
Yes for invidious, it's ok

it seems to be for piped.kavin.rocks that it doesn't work

**Reply by caywin25** [reply]
Hello,

As we can see on the screenshot, the redirect is done well to piped.kavin.rocks but when I display the video, it launches via youtube.

![Screenshot 2022-11-03 at 14-40-56 SearXNG](https://user-images.githubusercontent.com/55273439/199736921-666febb7-0cdc-469d-aa4a-4306b55b066c.png)

I test by calling:
`!yt crazy frog`

Thank you

**Reply by return42** [reply]
Strange .. seems there is something broken, here is how it looks like on my instance https://darmarit.org/searx/search?q=%21yt+crazy+frog

![grafik](https://user-images.githubusercontent.com/554536/199740225-4c79b04e-69d0-4851-b809-c3036b2f18c4.png)

**Reply by caywin25** [reply]
Yes but you it's with invidious.
And invidious is ok for me.

Just piped.kavin.rocks doesn't work :(
